api/src/main/java/org/apache/cloudstack/api/command/user/firewall/CreateFirewallRuleCmd.java

```diff
@@ -103,14 +105,13 @@ public String getProtocol() {
 
     @Override
     public List<String> getSourceCidrList() {
-        if (cidrlist != null) {
+        if (CollectionUtils.isNotEmpty(cidrlist) && (cidrlist.size() == 1 && StringUtils.isNotBlank(cidrlist.get(0)))) {
```

Comment:
I wonder if the check is correct 

maybe like
```
        if (CollectionUtils.isNotEmpty(cidrlist) && !(cidrlist.size() == 1 && StringUtils.isBlank(cidrlist.get(0)))) {
```
?

Reply:
Thanks @weizhouapache

Reply:
I'll add some unit tests to not make a mess of it